### PR TITLE
Add conditions to dhcp cleanup

### PR DESCRIPTION
**Pull request by czerw**
Cleanup must be done on qemu backends only and if wicked is running.

- Related ticket: https://progress.opensuse.org/issues/32716
- Needles: none
- Verification run:
SLES: http://10.100.12.105/tests/overview?distri=sle&groupid=33&version=15&build=%3Afix-shutdown
SLED: http://10.100.12.105/tests/overview?groupid=58&version=12-SP2&distri=sle&build=%3Afix-shutdown
openSUSE: http://10.100.12.105/tests/overview?version=Tumbleweed&distri=opensuse&build=%3Afix-shutdown

### CODE REVIEW

File: tests/shutdown/shutdown.pm [hunk 1]
```diff
@@ -31,9 +31,12 @@ sub run {
     if (!sle_version_at_least('12-SP2') && check_var('VIRTIO_CONSOLE', 1)) {
         add_serial_console('hvc0');
     }
+    # Shutdown network only on qemu backend
+    if (ckeck_var('BACKEND', 'qemu')) {
```
Comment (thehejik) [inline tests/shutdown/shutdown.pm]: typo check_var

File: tests/shutdown/shutdown.pm [hunk 2]
```diff
@@ -31,9 +31,12 @@ sub run {
...
+        systemctl 'stop network.service';
+        systemctl 'stop wickedd.service';
+    }
     # Clean dhcp ids from hdd image
-    systemctl 'stop network.service';
-    systemctl 'stop wickedd.service';
     assert_script_run('ls /var/lib/wicked/');
```
Comment (coolo) [inline tests/shutdown/shutdown.pm]: isn't SLED using NM and not wicked?

Comment (okurz) [reply tests/shutdown/shutdown.pm]: yes, same as openSUSE on a real laptop profile

Comment (czerw) [reply tests/shutdown/shutdown.pm]: You are right, I will add more checks.

Comment (czerw) [reply tests/shutdown/shutdown.pm]: @coolo @okurz I added more conditions ` if (!(get_var('FLAVOR', '') =~ /^Desktop/ || check_var('SLE_PRODUCT', 'sled')) || is_sle) {`

**Comment by czerw** [comment]
Verification links were updated. Dhcp cleanup is done on SLE and qemu backend only. SLED, openSUSE and SLES on non-qemu backends are skipped.

**Comment by okurz** [comment]
Your PR does not d really explain the reason or benefit. And why would it make sense to have this only for parts of our products?

**Comment by coolo** [comment]
I guess what we really want to check if wicked is present. How about doing this in a little longer script instead of not doing it on openSUSE?

if (qemu) {
  script_run('if test -d /var/lib/wicked/; then wicked stop; rm -f /var/lib/wicked/*xml; fi')
}

-> no need for further check_vars on products

**Comment by czerw** [comment]
@coolo Coded was updated. It is better to check wicked by network status, because directory `/var/lib/wicked` exists even network manager is running as primary. Verifications were re-done.

**Pull request merged**